akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -554,11 +553,45 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
       })
       sender() ! reply
 
-    case _: CurrentClusterState ⇒
-  }: Receive) orElse common
+  }: Receive).orElse[Any, Unit](common)
 
   def update[E <: DomainEvent](evt: E)(f: E ⇒ Unit): Unit
 
+  def watchStateActors(): Unit = {
+
+    // Optimization:
+    // Consider regions that don't belong to the current cluster to be terminated.
+    // This is an optimization that makes it operational faster and reduces the
+    // amount of lost messages during startup.
+    val nodes = Cluster(context.system).state.members.map(_.address)
```

Comment:
"we could use the same for Register instead of subscribing to MemberUp" - this exactly what I mean.